pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -62,11 +62,11 @@ var userRegistriesDir = filepath.FromSlash(".config/containers/registries.conf.d
 // The function errors if the newly created reference is not parsable.
 func (e *Endpoint) rewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
 	refString := ref.String()
-	if !refMatchesPrefix(refString, prefix) {
+	prefixLen := refMatchesPrefix(refString, prefix)
```

Comment:
A comment here would be good to describe that `refMatchesPrefix(...)` returns the length of the match.  Everything behind must be appended to the registries location.